types/object.go

```diff
@@ -0,0 +1,603 @@
+package types
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/NeowayLabs/abad/internal/utf16"
+)
+
+type (
+	// DataObject is a collection of named values.
+	DataObject struct {
+		// Class is the kind of object
+		class         string
+		notExtensible bool
+		props         map[string]*PropertyDescriptor
+	}
+
+	callable interface {
+		Call(this *DataObject, args []Value) Value
+	}
+)
+
+var (
+	// S is an alias to make utf-16 strings.
+	S            = utf16.S
```

Comment:
You could call it Str and drop the requirement for a comment. It is considerably smaller then utf16.S and a little more clear than S